modules/src/ics04_channel/handler/send_packet.rs

```diff
@@ -0,0 +1,267 @@
+use std::cmp::Ordering;
+
+use crate::events::IbcEvent;
+use crate::handler::{HandlerOutput, HandlerResult};
+use crate::ics02_client::state::ClientState;
+use crate::ics04_channel::channel::Counterparty;
+use crate::ics04_channel::channel::State;
+use crate::ics04_channel::events::SendPacket;
+use crate::ics04_channel::packet::{PacketResult, Sequence};
+use crate::ics04_channel::{context::ChannelReader, error::Error, error::Kind, packet::Packet};
+use crate::ics24_host::identifier::{ChannelId, PortId};
+use crate::Height;
+
+#[derive(Clone, Debug)]
+pub struct SendPacketResult {
+    pub port_id: PortId,
+    pub channel_id: ChannelId,
+    pub seq: Sequence,
+    pub seq_number: Sequence,
+    pub timeout_height: Height,
+    pub timeout_timestamp: u64,
+    pub data: Vec<u8>,
+}
+
+pub fn send_packet(ctx: &dyn ChannelReader, packet: Packet) -> HandlerResult<PacketResult, Error> {
+    let mut output = HandlerOutput::builder();
+
+    if packet.sequence.is_zero() {
+        return Err(Kind::ZeroPacketSequence.into());
+    }
+    if packet.timeout_height.is_zero() && packet.timeout_timestamp == 0 {
+        return Err(Kind::ZeroPacketTimeout.into());
+    }
+    if packet.data.is_empty() {
+        return Err(Kind::ZeroPacketData.into());
+    }
```

Comment:
This should be done in the `impl TryFrom<RawPacket> for Packet {...`
It will be needed later in `impl TryFrom<RawMsgRecvPacket> for MsgRecvPacket {` and all other messages that include a `Packet`, like acks, timeouts, etc.